### ISSUE TITLE
Use locally installed pug

### ISSUE THREAD

**Issue by benadamstyles** [yivo/pug-ruby#3]
Hey, I have also mentioned this in the other repo which I am using (https://github.com/DougBeney/jekyll-pug/issues/13) but is there any way to make `pug-ruby` use `pug` installed locally rather than globally? It is not working with `nvm`, and also installing globally [introduces other problems](https://www.sitepoint.com/solve-global-npm-module-dependency-problem/). If there's any way I can help solve this, please let me know! I'd love to help out if it is at all possible to resolve this issue.

**Comment by yivo** [comment]
Hi @Leeds-eBooks 

Yeah, this may be a problem.

pug-rails is designed in such way which allows to define multiple compiler types. I assume you want to use pug from `YOUR_PROJECT_ROOT/node_modules/pug`. If so it is very easy to implement. It can be achieved by adding something like `Pug::LocalCompiler` or `Pug::IsolatedCompiler` which are almost the same as `Pug::SystemCompiler`. Compiler will require `"#{Dir.pwd}/node_modules/pug"` to be present and load pug from it. 

This is good question. Thanks. If this fits your needs pls let me know so I can ship release tomorrow.

**Comment by benadamstyles** [comment]
Yes that is what I mean! Thanks. I think your solution sounds good, but as I am using https://github.com/DougBeney/jekyll-pug as an intermediary lib between my code and `pug-ruby`, it might be worth checking with @DougBeney to see if he can use this config option.

**Comment by DougBeney** [comment]
Very interesting. 

@yivo Do you think it would be a good idea to add a way to access this in pug-ruby, such as something like`Pug.use :project`? Then, in case someone has a complicated project setup, they could also define the project directory with something like `Pug.project_directory = "src/"`.
